adapters/appnexus/appnexus_test.go

```diff
@@ -207,6 +205,37 @@ func TestVideoTwoPods(t *testing.T) {
 	assert.NotEqual(t, adPodId1, adPodId2, "AdPod id should be different for different pods")
 }
 
+func TestVideoTwoPodsDifferentAdPodId(t *testing.T) {
```

Comment:
Please write this test using the json framework. I think it would be a good ides to add json tests to cover all true, all false, all missing, mix of true/missing, mix of false/missing, mix of true/false.

Reply:
Added json files to unit test cases you mentioned. I cannot find a good way to test 2 pods with adpodid enabled for both pods because I don't have a control on random number generation in unit tests. This specific case is covered in `TestVideoTwoPods` and `TestVideoTwoPodsManyImps`

Reply:
I suddenly started to get this error:
```
FAIL: TestVideoSinglePod (0.00s)
    appnexus_test.go:95: 
        	Error Trace:	appnexus_test.go:95
        	Error:      	Should be true
        	Test:       	TestVideoSinglePod
        	Messages:   	AdPod id doesn't present in Appnexus extension or has incorrect format
FAIL
```
It works if I run single test and doesn't if I run it using "validate.sh". RegExp matcher returns false. 
Please help!

Reply:
regMatch, matchErr := regexp.Match(`[0-9]19`, []byte(reqDataExt.Appnexus.AdPodId))
	assert.NoError(t, matchErr, "Regex match error should be nil")
	assert.True(t, regMatch, "AdPod id doesn't present in Appnexus extension or has incorrect format")

We are matching for an adpod ID that contains the digits '19', but aren't the first two digits of a `rand.Int63()` value. Why would we expect the adpod ID contain a one followed by a nine? It seems reasonably likely given the length of a 64 bit number, but certainly not expected.

Reply:
Regexp modified to `^[0-9]+$`